src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php

```diff
@@ -181,6 +181,14 @@ protected function compileAnsiOffset(Builder $query, $components)
 
         unset($components['orders']);
 
+        // As this moves the order statement to be part of the "select" statement, we need to
+        // move the bindings to the right position: right after "select".
+        $preferredBindingOrder = ['select', 'order', 'from', 'join', 'where', 'groupBy', 'having', 'union', 'unionOrder'];
```

Comment:
This should be defined in a class constant to avoid instantiating this array every time this method is called.

Reply:
Thanks for taking a look @spawnia ! Until we have a green light on the PR I prefer to keep it inline to make it easier to reason about.